### PR TITLE
More changes in templates

### PR DESCRIPTION
**Pull request by jesmrec**
Changes in the templates:

- In pull request template: 
  - Add release notes in advance to give time to be translated. Removed from issue template.

- In release template:
  - Move the signing tag to the end of the checklist, to assure there are no commits between the approved and the merge
  - Clarify that release branches will not be rebased anymore (direct merge instead)
  - New patch release section, with tasks for minor bugfixes 

Please @JuancaG05 and @Aitorbp review this and throw as many questions as needed to understand correctly the complete processes and why changes are suggested. Specially, the tasks in patch releases.

### CODE REVIEW

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some changes required and doubts here @jesmrec

File: .github/ISSUE_TEMPLATE/release_template.md
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:1]: Done

[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:1]: **About the first comment:**

The step ` [DIS] Update screenshots, if needed, in README.md` could be removed, because the other one you are mentioning covers it. Only one step for checking whether new screenshots are required is enough.

`store listing` refers to check whether screenshots are correctly listed in the Play Store. 

Summarizing:

- First, one step to check and generate screenshots if needed. It was moved to that upper position in order to generate the screenshots at the time QA passes, so after QA-approval it'd not be a blocker (assuming that generating screenshots requires some time)

- Second, step for uploading and publishing the generated screenshots

**About the second comment:**

It's correct for the Open Releases template, but not for Patch releases that are ad-hoc. When a patch release goes on, there is no release notes before. So, if you agree, i will re-write the step for open releases and let the patch releases as is. Is this OK?

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ## Related Issues
 App:
 
-- [ ] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
+- [ ] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
+- [ ] Add feature to Release Notes (if required)
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/pull_request_template.md:5]: Sure 👍  that's right.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 2]
```diff
@@ -33,15 +33,15 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
  - [ ] [QA] QA Approval
  - [ ] [DIS] Upload release APK and bundle to internal owncloud instance
  - [ ] [DOC] Ping in #documentation-internal that we are close to sign the new tags
- - [ ] [GIT] Create and sign tag `vM.m.p` in HEAD commit of release branch, in owncloud/android
- - [ ] [GIT] Move tag `latest` pointing the same commit as the release commit
  - [ ] [DIS] Upload & publish release bundle and changelog in Play Store
  - [ ] [DIS] Update screenshots and store listing, if needed, in Play Store
  - [ ] [GIT] Publish a new [release](https://github.com/owncloud/android/releases) in owncloud/android
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:38]: you're right, not posible to create release in GH if tag is not signed. Moving down...

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 3]
```diff
@@ -33,15 +33,15 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
...
  - [ ] [DIS] Release published in Play Store
  - [ ] [DIS] Publish post in central.owncloud.org ([`Category:News + Tag:android`](https://central.owncloud.org/tags/c/news/5/android))
  - [ ] [COM] Inform `#updates` and `#marketing` in internal chat that release is out
- - [ ] [GIT] Merge `release/M.m.p` branch into `master`, in owncloud/android
+ - [ ] [GIT] Create and sign tag `vM.m.p` in HEAD commit of release branch, in owncloud/android
+ - [ ] [GIT] Move tag `latest` pointing the same commit as the release commit
+ - [ ] [GIT] Merge (not rebase!!!) `release/M.m.p` branch into `master`, in owncloud/android
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:44]: Well, it's another option. I will reorder the tasks anyway.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 4]
```diff
@@ -53,6 +53,49 @@ Bugs & improvements:
 
 - [ ] (1) ...
 
+_____
+
+## Patch Release
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:59]: yep 👍

File: .github/ISSUE_TEMPLATE/release_template.md
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:1]: I will reword it to 

`[DIS] Update screenshots and check they are OK in Play Store`

[the remark this answers is not in the document]
Comment (JuancaG05) [reply .github/ISSUE_TEMPLATE/release_template.md:1]: **About the second comment:**

Ok, but it won't be common that the list of Release Notes is empty at this point since we always start from a recent stable version. I mean, we can remove the `...replacing emptyList with listOf...`, right? Just to avoid excess of text and information in these steps

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 5]
```diff
@@ -23,7 +24,7 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
  - [ ] [DIS] Create a folder for the new version like `M.m.p_YYYY-MM-DD` inside the `changelog` folder
  - [ ] [DIS] Move all changelog files from the `unreleased` folder to the new version folder
  - [ ] [DIS] Update screenshots, if needed, in README.md
- - [ ] [DEV] Add release notes replacing `emptyList` with `listOf` and adding inside `ReleaseNote()` with String resources, in `ReleaseNotesViewModel.kt`
+ - [ ] [DEV] Check release notes in `ReleaseNotesViewModel.kt` to assure nothing important is missing there.
```
Comment (JuancaG05) [inline .github/ISSUE_TEMPLATE/release_template.md:27]: Not important, but we don't have a final `.` in the other steps

File: .github/ISSUE_TEMPLATE/release_template.md
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:1]: I will set

` - [ ] [DEV] Add release notes to `ReleaseNotesViewModel.kt`

as generic, in fact, release notes will never be empty because we'll find always the release notes of the previously released version.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 6]
```diff
@@ -33,15 +34,15 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
  - [ ] [QA] QA Approval
  - [ ] [DIS] Upload release APK and bundle to internal owncloud instance
  - [ ] [DOC] Ping in #documentation-internal that we are close to sign the new tags
- - [ ] [GIT] Create and sign tag `vM.m.p` in HEAD commit of release branch, in owncloud/android
- - [ ] [GIT] Move tag `latest` pointing the same commit as the release commit
  - [ ] [DIS] Upload & publish release bundle and changelog in Play Store
  - [ ] [DIS] Update screenshots and store listing, if needed, in Play Store
+ - [ ] [GIT] Create and sign tag `vM.m.p` in HEAD commit of release branch, in owncloud/android
+ - [ ] [GIT] Move tag `latest` pointing the same commit as the release commit
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:40]: The way to do it is:

- First removing the existing tag
   - locally: `git tag --delete latest`
   - remote: `git push --delete origin latest`
   
- Then, create again the tag in the proper place and push it as we use to do

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 7]
```diff
@@ -53,6 +54,49 @@ Bugs & improvements:
...
+
+## Patch Release
+
+### TASKS:
+
+ - [ ] [GIT] Create branch `release/M.m.p` in owncloud/android from `latest`
+ - [ ] [DEV] Update version number and name in build.gradle in owncloudApp module
+ - [ ] [DEV] Update [SBOM](https://cloud.owncloud.com/f/6072870)
+ - [ ] [DIS] Create a folder for the new version like `M.m.p_YYYY-MM-DD` inside the `changelog` folder with the proper fixes
+ - [ ] [DIS] Move the `unreleased` folder from `master` to `changelog` folder, to avoid Calens' conflicts problems
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:67]: Yes, that's it. In this code review, [you'll find a deeper explanation](https://github.com/owncloud/android/pull/4331#discussion_r1504268804) about reasons to do this. If there is not `unreleased` in master yet, there is nothing to move, for sure.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 8]
```diff
@@ -53,6 +54,49 @@ Bugs & improvements:
...
+ - [ ] [QA] QA Approval
+ - [ ] [DIS] Upload release APK and bundle to internal owncloud instance
+ - [ ] [DIS] Upload & publish release bundle and changelog in Play Store
+ - [ ] [DIS] Update screenshots and store listing, if needed, in Play Store
+ - [ ] [GIT] Create and sign tag `vM.m.p` in HEAD commit of release branch, in owncloud/android
+ - [ ] [GIT] Move tag `latest` pointing the same commit as the release commit
+ - [ ] [GIT] Publish a new [release](https://github.com/owncloud/android/releases) in owncloud/android
+ - [ ] [DIS] Release published in Play Store
+ - [ ] [COM] Inform `#updates` in internal chat that release is out
+ - [ ] [GIT] Fix conflicts if they happen. GitHub option will help and will merge `master` into release branch, getting it ready to fast forward
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:83]: those two things are different. After signing the tag and so on, the final target is merging the release branch into `master`. We will be allowed to do this if release branch is updated with `master`. But, as we agreed, we will not rebase release branches anymore to avoid differences in the history, so , the only way to update the release branch with `master` is merging `master` into release branch fixing conflicts if there are. Github provides options to do this.

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 9]
```diff
@@ -53,6 +54,49 @@ Bugs & improvements:
...
+ - [ ] [GIT] Merge (not rebase!!!!) `release/M.m.p` branch into `master`, in owncloud/android
```
[the remark this answers is not in the document]
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:84]: The final step is merging into `master` as every PR, clicking the green button that we find in the PRs. `git merge master` is doable in the previous step, to update the release branch with `master`. You can do it locally and then push to the server, or using the GitHub tools as mentioned in the previous comment. 

Anyway, i will do a little change in the step description. Instead of `Merge (not rebase!!!!) ` i will set `Merge without rebasing` to clarify that merge and rebase are two different things, that we use to use as alternatives but actually they are not.

**Comment by jesmrec** [comment]
I will add additionally two steps to the OEM release template. In case a branch is needed for such release, it has to be merged as well

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 10]
```diff
@@ -23,7 +24,7 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
  - [ ] [DIS] Create a folder for the new version like `M.m.p_YYYY-MM-DD` inside the `changelog` folder
  - [ ] [DIS] Move all changelog files from the `unreleased` folder to the new version folder
  - [ ] [DIS] Update screenshots, if needed, in README.md
```
Comment (JuancaG05) [inline .github/ISSUE_TEMPLATE/release_template.md:26]: So this step is not needed anymore, right? We can merge it with `[DIS] Update screenshots and check they are OK in Play Store`. I would modify that step like I indicate in the next comment

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 11]
```diff
@@ -33,15 +34,15 @@ For OEM releases, keep the OEM Release template and remove the Open Release one
  - [ ] [QA] QA Approval
  - [ ] [DIS] Upload release APK and bundle to internal owncloud instance
  - [ ] [DOC] Ping in #documentation-internal that we are close to sign the new tags
+ - [ ] [DIS] Upload & publish release bundle and changelog in Play Store
+ - [ ] [DIS] Update screenshots and check they are OK in Play Store
```
Comment (JuancaG05) [inline .github/ISSUE_TEMPLATE/release_template.md:38]:
```suggestion
 - [ ] [DIS] Update screenshots and check they are OK in Play Store and in `README.md` (if needed)
```

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 7 above)
Comment (JuancaG05) [inline .github/ISSUE_TEMPLATE/release_template.md:67]:
```suggestion
 - [ ] [DIS] Copy the `unreleased` folder in `master` branch to `changelog` folder in this branch, to avoid Calens' conflicts problems
```
Maybe this tiny change improves understandability

File: .github/ISSUE_TEMPLATE/release_template.md [hunk 12]
```diff
@@ -53,6 +54,49 @@ Bugs & improvements:
...
+ - [ ] [DEV] Add release notes to `ReleaseNotesViewModel.kt` (in case patch release before OEM)
+ - [ ] [DIS] Check for new screenshots in Play Store / GitHub repo and generate them
+ - [ ] [DIS] Generate final bundle from last commit in owncloud/android
+ - [ ] [DEV] Code Review
+ - [ ] [QA] Test execution
+ - [ ] [QA] Trigger BitRise builds for unit tests and UI tests, in case CHANGELOG conflicts avoid them in GitHub
+ - [ ] [QA] QA Approval
+ - [ ] [DIS] Upload release APK and bundle to internal owncloud instance
+ - [ ] [DIS] Upload & publish release bundle and changelog in Play Store
+ - [ ] [DIS] Update screenshots and check they are OK in Play Store
```
Comment (JuancaG05) [inline .github/ISSUE_TEMPLATE/release_template.md:77]:
```suggestion
 - [ ] [DIS] Update screenshots and check they are OK in Play Store and in `README.md` (if needed)
```

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Some last tiny changes here @jesmrec

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 10 above)
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:26]: that was a leftover from other modifications. Yes, that step should be out from there.

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 7 above)
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:67]: 👍

File: .github/ISSUE_TEMPLATE/release_template.md (hunk 12 above)
Comment (jesmrec) [reply .github/ISSUE_TEMPLATE/release_template.md:77]: 👍

**Review by JuancaG05 — APPROVED** [review]
LGTM

**Review by Aitorbp — APPROVED** [review]
